src/ordered.rs

```diff
@@ -108,28 +108,30 @@ pub struct OrderedDocumentIterator<'a> {
     inner: linked_hash_map::Iter<'a, String, Bson>,
 }
 
+type DocumentMap<'a, T> = Map<OrderedDocumentIterator<'a>, fn((&'a String, &'a Bson)) -> T>;
```

Comment:
Clippy doesn't like the really long type name in the two structs below (which is fair), so I abstracted it into a type alias.